### PR TITLE
Update tutorials for easier usage

### PR DESCRIPTION
**Pull request by katrilh**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/power-ops-sdk/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [pyproject.toml](https://github.com/cognitedata/power-ops-sdk/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: docs/tutorials/advanced_cogshop_run.ipynb
Comment (katrilh) [inline docs/tutorials/advanced_cogshop_run.ipynb:64]: This part needs to be re-done

File: docs/tutorials/quickstart_cogshop_run.ipynb
Comment (katrilh) [inline docs/tutorials/quickstart_cogshop_run.ipynb:118]: Do not merge before this has been resolved

File: cognite/powerops/client/shop/cogshop_api.py
Comment (eriklien) [reply cognite/powerops/client/shop/cogshop_api.py:58]: Otherwise I think we should add optional arguments  `model_file, base_attribute_mappings, attribute_mapping_overrides, commands` etc.

File: cognite/powerops/client/shop/cogshop_api.py
Comment (eriklien) [inline cognite/powerops/client/shop/cogshop_api.py:110]: How will they be generated? (Related to comment above.)

File: cognite/powerops/client/shop/cogshop_api.py
Comment (eriklien) [inline cognite/powerops/client/shop/cogshop_api.py:170]:
```suggestion
        return self.prepare_shop_case_with_existing_scenario(shop_file_list, start_time, end_time, scenario_write, case_external_id)

    def prepare_shop_case_with_existing_scenario(
        self,
        shop_file_list: list[tuple[str, str, bool, str]],
        start_time: datetime.datetime,
        end_time: datetime.datetime,
        scenario: str | ShopScenarioWrite,
```
(not a complete suggestion)
To save some duplicate code.

File: cognite/powerops/client/shop/cogshop_api.py
Comment (eriklien) [inline cognite/powerops/client/shop/cogshop_api.py:226]: **suggestion/comment**: This one seems unnecessary - can't we just do `upsert` instead of `write_resources`?

File: cognite/powerops/client/shop/cogshop_api.py
Comment (eriklien) [inline cognite/powerops/client/shop/cogshop_api.py:230]: **suggestion/comment**: Seems unnecessary

File: cognite/powerops/client/shop/cogshop_api.py
Comment (eriklien) [inline cognite/powerops/client/shop/cogshop_api.py:246]: **suggestion/comment**: These also seem unnecessary...

**Review by eriklien — COMMENTED** [review]
Have reviewed the SDK changes (not the notebooks yet) - added some suggestions/questions/comments.

File: docs/tutorials/quickstart_cogshop_run.ipynb
Comment (eriklien) [inline docs/tutorials/quickstart_cogshop_run.ipynb:299]:
```suggestion
    "Notice that the retrieved case is not suffixed with `Write` \n",
```

File: cognite/powerops/client/shop/cogshop_api.py
Comment (katrilh) [reply cognite/powerops/client/shop/cogshop_api.py:58]: It was only used for a small part of the advanced guide. I will take it out of `cogshop_api.py` and instead use the generated sdk's object directly. 

Otherwise this file just becomes manually written wrappers for the already generated sdk

File: cognite/powerops/client/shop/cogshop_api.py
Comment (katrilh) [reply cognite/powerops/client/shop/cogshop_api.py:92]: Yes, setting the external id to `None` sets off pydantic's validation detection, not passing it at all lets them use the external id factory as defined on the `DomainModelWrite` level. 
I agree it's not very pretty to look at and i wish it wasnt this way

File: cognite/powerops/client/shop/cogshop_api.py
Comment (eriklien) [inline cognite/powerops/client/shop/cogshop_api.py:209]: **Comment**: Yes, that would be great (but maybe not in this PR)

File: cognite/powerops/client/shop/cogshop_api.py
Comment (eriklien) [inline cognite/powerops/client/shop/cogshop_api.py:212]: Feels very strange to have this that does just the same as `retrieve_shop_case`.
If we keep it as a separate function, I would like the docstring to explain when it may be better to use this one (if the reason is something else than that the other one is unstable - if it is just for the instability, maybe use graphql as fallback inside `retrieve_shop_case` instead?)

File: docs/tutorials/quickstart_cogshop_run.ipynb
Comment (eriklien) [inline docs/tutorials/quickstart_cogshop_run.ipynb:1]: Looks good. I'd consider removing the graphql functions that do (almost) the same as "non-graphQL" functions, even if we keep both in the SDK (ref comment above) in the quickstart guide (and maybe mention that in the advanced guide, if we keep both).

File: docs/tutorials/advanced_cogshop_run.ipynb
Comment (eriklien) [inline docs/tutorials/advanced_cogshop_run.ipynb:1]: **request**: Remove duplicate title "### Step 5: View the ShopResult generated

Comment (eriklien) [reply docs/tutorials/advanced_cogshop_run.ipynb:1]: **suggestion** Change title from "## Example 2: Reusing a ShopScenario used to set shop version" to "## Example 2: Reusing a ShopScenario (used to set SHOP version)" or even just "## Example 2: Reusing ShopScenario instance"
(Trying to be consistent with casing of SHOP - not shop + less/no emphasis on the purpose of the scenario, which is not the purpose of reusing it)

Comment (eriklien) [reply docs/tutorials/advanced_cogshop_run.ipynb:1]: **suggestion**: Change title from "### Step 1: Create reusable ShopScenarios with ShopScenarios" to "### Step 1: Create reusable ShopScenario (with ShopModel) instances" (or something else that make more sense)

Comment (eriklien) [reply docs/tutorials/advanced_cogshop_run.ipynb:1]: **suggestion**: Change comment "# This will fail since we did not upsert the scenario" to "# This will fail because the scenario does not exist in CDF (we did not upsert it)" (or something like that)

Comment (eriklien) [reply docs/tutorials/advanced_cogshop_run.ipynb:1]: **suggestion**: Near comment "# However, if we reference the Write object in the case creation it will work", mention that it will be created/upserted/written to CDF

Comment (eriklien) [reply docs/tutorials/advanced_cogshop_run.ipynb:1]: **suggestion**: Where downloading pre_run files, call the temporary files pre_run.yaml instead of log_file.log - I know the file names are not used for anything, but I still got confused by this because the per_run file is not a log file (but messages and cplex log are log files)
**nitpick**: I assume that this global_options thing will not be added if we only use one SHOP file (a complete case, not a separate commands file), so the example (or our implementation?) can be a bit confusing (I don't think you have to change that, though)

Comment (eriklien) [reply docs/tutorials/advanced_cogshop_run.ipynb:1]: But all in all, it looks good.

**Review by eriklien — APPROVED** [review]
Have added some requests/suggestions, please consider those before merging, but I think things are in a good state now (and I trust you will do the relevant changes), so it's (finally) approved.

File: cognite/powerops/client/shop/cogshop_api.py
Comment (katrilh) [reply cognite/powerops/client/shop/cogshop_api.py:212]: Updated docstring

File: cognite/powerops/client/shop/cogshop_api.py
Comment (eriklien) [inline cognite/powerops/client/shop/cogshop_api.py:215]: I guess that's the truth, but do we really want to highlight that the other method is unstable?

Comment (eriklien) [inline cognite/powerops/client/shop/cogshop_api.py:215]:
```suggestion
        This is provided as a fallback for cases when retrieve_shop_case does not perform well"""
```

File: docs/tutorials/advanced_cogshop_run.ipynb
Comment (eriklien) [inline docs/tutorials/advanced_cogshop_run.ipynb:408]:
```suggestion
    "See details in the docstring of `retrieve_shop_case_graphql`"
```

File: cognite/powerops/client/shop/cogshop_api.py
Comment (eriklien) [inline cognite/powerops/client/shop/cogshop_api.py:225]:
```suggestion
        This is provided as a fallback option for cases when `list_shop_results_for_case` does not perform well"""

```

**Review by eriklien — APPROVED** [review]

**Pull request merged**
